### ISSUE TITLE
`no-member` when creating an object through an inherited factory method

### ISSUE THREAD

**Issue by carmenbianca** [pylint-dev/pylint#9203]
### Bug description

```python
# pylint: disable=missing-module-docstring,missing-class-docstring,disallowed-name,unused-argument,too-few-public-methods,missing-function-docstring

class Base:
    def __init__(self):
        self.foo = "bar"

    @classmethod
    def from_dict(cls, values):
        return cls()

class Child(Base):
    def __init__(self):
        super().__init__()
        self.baz = "quz"

    @classmethod
    def from_dict(cls, values):
        return super().from_dict(values)

child = Child.from_dict({})
print(child.baz)
```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
************* Module a
a.py:21:6: E1101: Instance of 'Base' has no 'baz' member (no-member)

-----------------------------------
Your code has been rated at 6.15/10
```

### Expected behavior

`no-member` is not a valid error here.

### Pylint version

```shell
pylint 2.17.7
astroid 2.15.8
Python 3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0]
```

### OS / Environment

Debian 12

**Comment by carmenbianca** [comment]
Real-world example:

https://codeberg.org/carmenbianca/protokolo/src/commit/358fe0d04dadef1857d1e937506a97ab7e3824dd/tests/test_config.py#L282

**Comment by Markovvn1w** [comment]
A slightly different way to reproduce the same (probably) problem:
```python
from typing import Tuple

class AnotherClass:
    @classmethod
    def create(cls) -> Tuple["AnotherClass"]:
        return (cls(),)

    def get(self) -> None:
        pass

class MainClass:
    _test: AnotherClass

    def __init__(self) -> None:
        (self._test,) = AnotherClass.create()
        self._test.get()  # E1101: Instance of 'MainClass' has no 'get' member (no-member)
```
